src/components/Input/index.js

```diff
@@ -1,81 +1,139 @@
+    ]),
+    help: PropTypes.string,
+    inverted: PropTypes.bool,
+    label: PropTypes.string,
+    name: PropTypes.string,
+    value: PropTypes.string.isRequired,
+    onBlur: PropTypes.func,
+    onChange: PropTypes.func,
+    onFocus: PropTypes.func,
   }
 
   static defaultProps = {
-    type: 'text',
-    fullWidth: true,
-    disabled: false,
+    value: '',
+  }
+
+  state = {
+    focused: false,
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.input = React.createRef()
   }
 
-  onChange = ({ target: { value } }) => {
-    const { onChange } = this.props
-    onChange(value)
+  onFocus = () => {
+    const { onFocus } = this.props
+
+    this.setState({ focused: true })
+
+    if (onFocus) {
+      onFocus()
+    }
+  }
+
+  onBlur = () => {
+    const { onBlur } = this.props
+
+    this.setState({ focused: false })
+
+    if (onBlur) {
+      onBlur()
+    }
+  }
+
+  focus = () => {
+    this.input.current.focus()
   }
 
   render () {
     const {
-      value,
-      type,
-      placeholder,
-      label,
-      style,
-      fullWidth,
+      append,
       error,
+      help,
+      inverted,
       disabled,
+      label,
+      name,
+      prepend,
+      value,
       onChange,
-      warning,
       ...props
     } = this.props
-    const showLabel = label && placeholder && value
-    const inputClassNames = cn(
-      'input-field__input',
-      { 'input-field__input--with-label': showLabel },
-      { 'input-field__input--full-width': fullWidth },
-      { 'input-field__input--error': error },
-      { 'input-field__input--warning': warning },
-      { 'input-field__input--disabled': disabled },
-    )
-    const labelClassNames = cn(
-      'input-field__label',
-      { 'input-field__label--hide': !showLabel },
-    )
+
+    const {
+      focused,
+    } = this.state
+
+    const classes = [
+      'mc-form-input',
+      focused ? 'mc-form-input--focus' : '',
+      label ? '' : 'mc-form-input--no-label',
+      value ? 'mc-form-input--modified' : '',
+      inverted ? 'mc-form-input--invert' : '',
+      disabled ? 'mc-form-input--disabled' : '',
+      error ? 'mc-form-input--error' : '',
+    ].join(' ')
 
     return (
-      <div style={style} className='input-field'>
-        {label &&
-          <label className={labelClassNames}>{label}</label>
-        }
-        {error &&
-          <span className='input-field__error'>{error}</span>
-        }
-        {warning &&
-          <span className='input-field__warning'>{warning}</span>
+      <Fragment>
+        <div
+          className={classes}
+          onClick={this.focus}
+        >
+          {prepend &&
+            <div className='mc-form-prepend'>
+              {prepend}
+            </div>
+          }
+
+          <div className='mc-form-input__input'>
+            <input
+              name={name}
+              type='text'
+              value={value}
+              disabled={disabled}
+              {...props}
```

Comment:
we use this in case you want to pass in other input props, like `maxlength` and turning off autofill.  we should allow people to pass along any props without us having to define them all

Reply:
where is ...props coming from, shouldn't that be undefined

Reply:
oh wait, is that what `...props` is in the deconstructing?

Reply:
Yep